Mini Challenge 3 - Hooks

asosacas opened the pull request:
# What's new?
- Adding video detail screen to watch the video and see the recommendations based on said video.
# Test Coverage Results
![Screen Shot 2021-03-04 at 23 01 28](https://user-images.githubusercontent.com/69918929/110131066-5f717b00-7d8f-11eb-93dd-9efb944c2be2.png)

erickwize commented on src/hooks/resetScroll/resetScroll.hook.jsx:
```diff
@@ -0,0 +1,13 @@
+
+const useResetScroll = () => {
+  const resetScrollRef = useRef(null);
+  useEffect(() => {
+    if (resetScrollRef.current) {
+      resetScrollRef.current.scrollLeft = 0;
+      resetScrollRef.current.scrollTop = 0;
+    }
+  });
```
Nice hook, but if you add some interactions that trigger the rerender the hook will be called again and it will scroll to the top. I can see on the working app why did you scroll to top-left, but I recommend using the dependency with `[]` so it should be triggered only the first time

erickwize commented on src/api/index.js:
```diff
@@ -1,882 +1,22 @@
+};
+
+export default {
+  searchVideos: (searchValue) => {
+    return fetch(apiUrlBuilder('search', { q: searchValue })).then((response) =>
+      response.json()
+    );
+  },
+  getRelatedVideos: (videoId) => {
+    return fetch(
+      apiUrlBuilder('search', { relatedToVideoId: videoId })
+    ).then((response) => response.json());
```
You could remove the `return`, but, if you believe it is more clear with the return there it is good 😊

erickwize commented on src/components/Header/Header.component.jsx:
```diff
@@ -1,19 +1,28 @@
-  </StyledNavBar>
-);
+const Header = ({ setSearch }) => {
+  const [tempSearch, setTempSearch] = useState();
+  const onSubmit = (ev) => {
+    setSearch(tempSearch);
+    ev.preventDefault();
+  };
+  return (
+    <StyledNavBar>
+      <StyledNavBar.Left>
+        <FaBars />
+        <form onSubmit={onSubmit}>
+          <InputText onChange={setTempSearch} />
+        </form>
```
I recommend using [controlled components](https://dmitripavlutin.com/controlled-inputs-using-react-hooks/), it is not necessary because you're solving the issue with uncontrolled components, but, it is a better practice to have controlled components.

```js
<InputText onChange={(event) => setTempSearch(event.target.value)}  value={tempSearch}/>
```

erickwize reviewed (approved):
## Comments
Nice work! Keep it with those tests! I saw the custom-created hooks, nice work there! I left some comments and recommendations, but, LGTM 👍 

I see that the HTML characters are not being parsed (this is not evaluated) but it could be improved. [(Probable solution)](https://shripadk.github.io/react/docs/jsx-gotchas.htmlz)
![image](https://user-images.githubusercontent.com/55102799/110730495-29e0ed80-81e6-11eb-8d6a-76d4acd4aa44.png)


## Acceptance Criteria
- [x] Videos in the Home View are fetched from the YouTube API and displayed correctly.
- [x] Users can search for YouTube videos using the search field on the header.
- [x] A video can be played from the **Video Details View** after clicking on it.
- [x] The video information and related videos list are displayed correctly on the **Video Details View**.
- [x] When a user clicks on a related video the video player, information and related videos list are updated.

### Bonus Points
- [x] Custom Hooks for API calls are implemented and tested correctly.
- [x] Testing coverage is above 60%. (Please include a screenshot of the code coverage report in your PR description).

asosacas commented:
@erickwize Thank you, appreciating the feedback!
I've made changes according to the comments, they all make sense to me 👍 
- Added dependency on useEffect hook
- Cleaning unneeded return 
- Making InputText a controlled component. (Also removing that shortcut I added for the onChange handling, I think it just making it confusing)
